### PR TITLE
feat: add `ST06` rule

### PR DESCRIPTION
**Pull request by gvozdvmozgu**

### CODE REVIEW

**Review by benfdking — APPROVED** [review]
LFG!

**Comment by benfdking** [comment]
@gvozdvmozgu  good to go?

File: crates/lib/src/rules/structure/ST06.rs [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+
+fn validate(
+    i: usize,
+    segment: ErasedSegment,
+    current_element_band: &mut Option<usize>,
+    violation_exists: &mut bool,
+    seen_band_elements: &mut [Vec<ErasedSegment>],
+) {
+    if seen_band_elements[i + 1..] != vec![Vec::new(); seen_band_elements[i + 1..].len()] {
+        dbg!(segment.raw());
```
Comment (nikolaevx) [inline crates/lib/src/rules/structure/ST06.rs]:
```suggestion

```

**Comment by gvozdvmozgu** [comment]
yes

**Pull request merged**